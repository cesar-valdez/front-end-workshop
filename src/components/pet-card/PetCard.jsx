import React from 'react';
import PetCardSchema from './PetCard.schema';

import './pet-card.scss';

const PetCard = ({ id, img, name, details }) => (
  <article className="yta_PetCard">
    <div className="yta_PetCard-img" style={{ backgroundImage: `url(${img})` }} />
    <div className="yta_PetCard-description">
      <h3 className="yta_PetCard-description-name"> {name}</h3>
      <p className="yta_PetCard-description-details"> {details} </p>
      <p className="yta_PetCard-description-link">
        <a href={`pet/${id}`} className="yta_Link-outline--primary">
          Adoptar
        </a>
      </p>
    </div>
  </article>
);

PetCard.prototypes = PetCardSchema;

export default PetCard;
