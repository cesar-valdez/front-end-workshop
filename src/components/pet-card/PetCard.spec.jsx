import React from 'react';
import { shallow } from 'enzyme';

import PetCard from './PetCard';

describe('PetCard', () => {
  const petCard = shallow(<PetCard />);

  it('should render PetCard componente', () => {
    expect(petCard).toMatchSnapshot();
  });
});
