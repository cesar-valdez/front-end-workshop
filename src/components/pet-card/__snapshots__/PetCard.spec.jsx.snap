// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetCard should render PetCard componente 1`] = `
<article
  className="yta_PetCard"
>
  <div
    className="yta_PetCard-img"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
  <div
    className="yta_PetCard-description"
  >
    <h3
      className="yta_PetCard-description-name"
    >
       
    </h3>
    <p
      className="yta_PetCard-description-details"
    >
       
       
    </p>
    <p
      className="yta_PetCard-description-link"
    >
      <a
        className="yta_Link-outline--primary"
        href="pet/undefined"
      >
        Adoptar
      </a>
    </p>
  </div>
</article>
`;
